src/Utils/ArrayDiff.php

```diff
@@ -48,4 +50,39 @@ public function diff($oldData, $newData)
 
         return $diff;
     }
+
+    /**
+     * Compare the type and the property values of two objects.
+     * Return true if they are the same, false otherwise.
+     * If the type is the same and all properties are the same, this will return true, even if they are not the same instance.
+     */
+    public function compareObjects(object $object1, object $object2): bool
+    {
+        // Check if the objects are of the same type.
+        $obj1Class = $object1::class;
+        $obj2Class = $object2::class;
+        if ($obj1Class !== $obj2Class) {
```

Comment:
```suggestion
        if ($object1::class !== $object2::class) {
```
The only concern I have regarding this is about class aliases, but I don't know if that is valid use case for this method.

Reply:
I think class aliases are out of scope of how the EntityAuditBundle is used.